### DISCUSSION TITLE
Can I use vcpkg artifacts with private repository?

### DISCUSSION THREAD

**Discussion by Sn1cket** [microsoft/vcpkg#33725] [Q&A]
I would like to use vcpkg artifacts to aquire the tools for building and debugging a project.

Is it possible to use a private repository that needs authentication (e.g. a GitLab repository) in a vcpkg-configuration.json file?

Like this:
```
{
  "registries": [
    {
      "name": "myregistry",
      "location": "https://url/to/private/repo/archive.zip",
      "kind": "artifact"
    }
  ]
}
```

**Comment by luncliff** [comment]
Maybe you can put auth token in "location" field.
I solved private Git repo access with the following `vcpkg-configuration.json`

Hope it also works for "artifact" kinds.

```json
{
  "default-registry": null,
  "registries": [
    {
      "packages": [
        "zlib-ng"
      ],
      "kind": "git",
      "repository": "https://ghp_abcd1234@private.github-enterprise.com/my-team/vcpkg-registry",
      "baseline": "000000"
    }
  ]
}
```

**Comment by DimRochette** [comment]
If you store the token on git credential side it works on CI/docker 
```
# Set up git credentials for private repo access
RUN git config --global credential.helper store
RUN echo "https://x-oauth-basic:$PRIVATE_GITHUB_TOKEN@github.com" > $NAME_OF_USED_HOME/.git-credentials
```

**Reply by ethanmusser** [reply]
This is a good solution, something like this should be called out in the documentation.
